src/renderers/shared/reconciler/ReactChildReconciler.js

```diff
@@ -97,7 +99,8 @@ var ReactChildReconciler = {
         nextChildren[name] = prevChild;
       } else {
         if (prevChild) {
-          ReactReconciler.unmountComponent(prevChild, name);
+          removedChildren[name] = ReactReconciler.getNativeNode(prevChild);
```

Comment:
I think the general idea I had for ReactChildReconciler would be that the delete hooks of the renderer could be called directly here. That way the "enqueue delete" call could be called before the unmount without a queue.